src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/talismans/Talisman.java

```diff
@@ -130,48 +134,61 @@ private static boolean hasMessage(Talisman talisman) {
         return talisman.getMessageSuffix() != null;
     }
 
-    public static boolean checkFor(Event e, SlimefunItemStack stack) {
+    public static ItemStack checkFor(@Nonnull Event e, @Nonnull SlimefunItemStack stack) {
         return checkFor(e, stack.getItem());
     }
 
-    public static boolean checkFor(Event e, SlimefunItem item) {
+    @Nullable
+    public static ItemStack checkFor(@Nonnull Event e, @Nonnull SlimefunItem item) {
         if (!(item instanceof Talisman)) {
-            return false;
+            return null;
         }
 
         Talisman talisman = (Talisman) item;
         if (ThreadLocalRandom.current().nextInt(100) > talisman.getChance()) {
-            return false;
+            return null;
         }
 
         Player p = getPlayerByEventType(e);
         if (p == null || !pass(p, talisman)) {
-            return false;
+            return null;
         }
 
         ItemStack talismanItem = talisman.getItem();
+        ItemStack enderTalisman = talisman.getEnderVariant();
+
+        if (!Slimefun.hasUnlocked(p, talisman, true)) {
+            return null;
+        }
 
         if (SlimefunUtils.containsSimilarItem(p.getInventory(), talismanItem, true)) {
-            if (Slimefun.hasUnlocked(p, talisman, true)) {
-                activateTalisman(e, p, p.getInventory(), talisman);
-                return true;
-            } else {
-                return false;
-            }
-        } else {
-            ItemStack enderTalisman = talisman.getEnderVariant();
-
-            if (SlimefunUtils.containsSimilarItem(p.getEnderChest(), enderTalisman, true)) {
-                if (Slimefun.hasUnlocked(p, talisman, true)) {
-                    activateTalisman(e, p, p.getEnderChest(), talisman);
-                    return true;
-                } else {
-                    return false;
-                }
-            } else {
-                return false;
+            activateTalisman(e, p, p.getInventory(), talisman);
+            return retrieveTalismanFromInventory(p.getInventory(), talisman);
+        }
+        else if (SlimefunUtils.containsSimilarItem(p.getEnderChest(), enderTalisman, true)) { 
+            activateTalisman(e, p, p.getEnderChest(), talisman);
+            return retrieveTalismanFromInventory(p.getEnderChest(), talisman);
+        }
+
+        return null;
+    }
+
+    @Nullable
+    private static ItemStack retrieveTalismanFromInventory(@Nonnull Inventory inv, @Nonnull Talisman talisman) {
+        ItemStack[] contents = inv.getContents();
+        for (int i = 0; i < contents.length; i++) {
+            ItemStack item = contents[i];
+
+            if (SlimefunUtils.isItemSimilar(item, talisman.getItem(), true, false)) {
```

Comment:
Also can just do this...

```suggestion
            if (talisman.isItem(item)) {
```